tools/cd_scripts/package_gcsfuse.sh

```diff
@@ -0,0 +1,85 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script will create gcsfuse package for both amd64 and arm64 machines.
+# After creation of package we are uploading it on gcs bucket.
+
+# The script is taking around 20-25 minutes on n2-standard-32 machine.
+set -e
+# create-gcsfuse-packages VM is a fixed VM in GCP project gcs-fuse-test
+# Fetch metadata value of the key "RELEASE_VERSION"
+gcloud compute instances describe create-gcsfuse-packages --zone us-west1-b --flatten="metadata[RELEASE_VERSION]" >>  release_version.txt
+# cat release_version.txt
+# ---
+#   v1.0.7
+x=$(sed '2!d' release_version.txt)
+# echo "$x"
+#   v1.0.7(contains preceding spaces)
+RELEASE_VERSION=$(echo "$x" | sed 's/[[:space:]]//g')
+# Remove spaces
+# release version
+# echo $RELEASE_VERSION
+# v1.0.7
+rm release_version.txt
+echo "$RELEASE_VERSION"
+
+# Fetch metadata value of the key "UPLOAD_BUCKET"
+gcloud compute instances describe create-gcsfuse-packages --zone us-west1-b --flatten="metadata[UPLOAD_BUCKET]" >>  bucket_name.txt
+# cat bucket_name.txt
+# ---
+#   bucket name
+x=$(sed '2!d' bucket_name.txt)
+# echo "$x"
+#   bucket_name(contains preceding spaces)
+# Remove spaces
+UPLOAD_BUCKET=$(echo "$x" | sed 's/[[:space:]]//g')
+echo BUCKET NAME="$UPLOAD_BUCKET"
+rm bucket_name.txt
+
+sudo apt-get update
+echo Install docker
+sudo apt install apt-transport-https ca-certificates curl software-properties-common -y
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu focal stable"
+apt-cache policy docker-ce
+sudo apt install docker-ce -y
+echo Install git
+sudo apt-get install git -y
+# It is require for multi-arch support
+sudo apt-get install qemu-user-static binfmt-support
+
+git clone https://github.com/GoogleCloudPlatform/gcsfuse.git
+cd gcsfuse/tools/package_gcsfuse_docker/
+echo "Building docker for amd64 ..."
+sudo docker buildx build --load . -t gcsfuse-release-amd64:"$RELEASE_VERSION" --build-arg GCSFUSE_VERSION="$RELEASE_VERSION" --build-arg ARCHITECTURE=amd64 --platform=linux/amd64 &> docker_amd64.log &
+echo "Building docker for arm64 ..."
+# It is necessary for cross-platform image building because it creates a builder instance that is capable of
+# building images for multiple architectures.
+sudo docker buildx create --name mybuilder --bootstrap --use
+sudo docker buildx build --load . -t gcsfuse-release-arm64:"$RELEASE_VERSION" --build-arg GCSFUSE_VERSION="$RELEASE_VERSION" --build-arg ARCHITECTURE=arm64 --platform=linux/arm64 &> docker_arm64.log &
+echo "Waiting for both builds to complete ..."
+wait "$!"
+if [[ $? -ne 0 ]]; then
```

Comment:
Unsure if we need this, since, `set -e` will exit immediately if a command exits with a non-zero status.
Can you verify?

Reply:
Maybe set +e before running the docker commands so that we don't lose logs.

Reply:
Yes Thanks